### PR TITLE
Use native CMake variables for compiler flags

### PR DESCRIPTION
**Pull request by upsj**
Pulled out of #1334, this uses the natively supported `CMAKE_<LANG>_FLAGS` variables instead of our custom variables and sets them in CI instead of CMake, which is discouraged in its genericity, since `-Wpedantic` is only supported in gcc-like compilers.

The custom variables are not necessary, because they can instead be handled by appropriately scoped CMake variables

TODO:
- [x] update README instructions for SYCL

### CODE REVIEW

**Review by MarcelKoch — COMMENTED** [review]
I guess we could think about adding cmake presets. That would be an easy way to keep the `-Wpedantic`. I was already looking into that, so I will get something ready for discussion.

File: CMakeLists.txt [hunk 1]
```diff
@@ -51,20 +51,6 @@ option(GINKGO_BENCHMARK_ENABLE_TUNING
...
-elseif(GINKGO_BUILD_SYCL OR CMAKE_CXX_COMPILER MATCHES "dpcpp|icpx")
-    # For now always use `-ffp-model=precise` with DPC++. This can be removed when
-    # the floating point issues are fixed.
-    set(GINKGO_COMPILER_FLAGS "-Wpedantic;-ffp-model=precise" CACHE STRING
-        "Set the required CXX compiler flags, mainly used for warnings. Current default is `-Wpedantic;-ffp-model=precise`")
-else()
-    set(GINKGO_COMPILER_FLAGS "-Wpedantic" CACHE STRING
-        "Set the required CXX compiler flags, mainly used for warnings. Current default is `-Wpedantic`")
-endif()
-set(GINKGO_CUDA_COMPILER_FLAGS "" CACHE STRING
```
[the remark this answers is not in the document]
Comment (upsj) [reply CMakeLists.txt:66]: There is no native equivalent until we tackle #1334

File: dpcpp/CMakeLists.txt [hunk 2]
```diff
@@ -98,7 +98,6 @@ target_compile_options(ginkgo_dpcpp PRIVATE "${GINKGO_DPCPP_FLAGS}")
 # If we would like to use SOURCES, please use the new files copied from GKO_UNIFIED_COMMON_SOURCES.
 # Otherwise, the source's properties will be changed by add_sycl_to_target
 gko_add_sycl_to_target(TARGET ginkgo_dpcpp)
-target_compile_options(ginkgo_dpcpp PRIVATE "${GINKGO_COMPILER_FLAGS}")
```
[the remark this answers is not in the document]
Comment (upsj) [reply dpcpp/CMakeLists.txt:101]: I was considering it, but that could lead to weird situations where some math functions might have two different definitions depending on whether they are compiled as part of `libginkgo_dpcpp.so` or from `ginkgo/core/base/math.hpp`.

File: reference/get_info.cmake [hunk 3]
```diff
@@ -1,4 +1,3 @@
-ginkgo_print_module_header(${detailed_log} "Reference")
-ginkgo_print_variable(${detailed_log} "GINKGO_BUILD_REFERENCE")
-ginkgo_print_variable(${detailed_log} "GINKGO_COMPILER_FLAGS")
-ginkgo_print_module_footer(${detailed_log} "")
+#ginkgo_print_module_header(${detailed_log} "Reference")
+#ginkgo_print_variable(${detailed_log} "GINKGO_BUILD_REFERENCE")
+#ginkgo_print_module_footer(${detailed_log} "")
```
Comment (MarcelKoch) [inline reference/get_info.cmake:3]: Uncomment?

File: omp/get_info.cmake [hunk 4]
```diff
@@ -2,8 +2,7 @@ ginkgo_print_module_header(${detailed_log} "OpenMP")
 ginkgo_print_variable(${detailed_log} "OpenMP_CXX_FLAGS")
 ginkgo_print_variable(${detailed_log} "OpenMP_CXX_LIB_NAMES")
 ginkgo_print_variable(${detailed_log} "OpenMP_CXX_LIBRARIES")
-ginkgo_print_module_footer(${detailed_log} "OMP variables:")
-ginkgo_print_variable(${detailed_log} "GINKGO_COMPILER_FLAGS")
+#ginkgo_print_module_footer(${detailed_log} "OMP variables:")
```
Comment (MarcelKoch) [inline omp/get_info.cmake:5]: same?

**Review by MarcelKoch — CHANGES_REQUESTED** [review]

**Review by MarcelKoch — APPROVED** [review]

**Review by thoasm — APPROVED** [review]
Looks good to me!

**Pull request merged**